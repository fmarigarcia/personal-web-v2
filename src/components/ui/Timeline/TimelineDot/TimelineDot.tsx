import React from 'react';
import { clsx } from 'clsx';

interface TimelineDotProps {
    isSelected: boolean;
}

export const TimelineDot: React.FC<TimelineDotProps> = ({ isSelected }) => {
    return (
        <div
            className={clsx(
                'w-4 h-4 rounded-full border-3 transition-all duration-300 mb-4',
                {
                    'bg-blue-600 border-blue-600 scale-125 shadow-lg':
                        isSelected,
                    'bg-white border-blue-600 group-hover:bg-blue-100 group-hover:scale-110':
                        !isSelected,
                }
            )}
        />
    );
};
