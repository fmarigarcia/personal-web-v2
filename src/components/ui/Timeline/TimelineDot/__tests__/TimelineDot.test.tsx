import { render } from '@testing-library/react';
import { TimelineDot } from '../TimelineDot';

describe('TimelineDot', () => {
    it('should apply selected styles when isSelected is true', () => {
        const { container } = render(<TimelineDot isSelected={true} />);

        const dot = container.firstChild as HTMLElement;
        expect(dot).toHaveClass(
            'bg-blue-600',
            'border-blue-600',
            'scale-125',
            'shadow-lg'
        );
    });

    it('should apply unselected styles when isSelected is false', () => {
        const { container } = render(<TimelineDot isSelected={false} />);

        const dot = container.firstChild as HTMLElement;
        expect(dot).toHaveClass(
            'bg-white',
            'border-blue-600',
            'group-hover:bg-blue-100',
            'group-hover:scale-110'
        );
    });

    it('should have correct base classes', () => {
        const { container } = render(<TimelineDot isSelected={false} />);

        const dot = container.firstChild as HTMLElement;
        expect(dot).toHaveClass(
            'w-4',
            'h-4',
            'rounded-full',
            'border-3',
            'transition-all',
            'duration-300',
            'mb-4'
        );
    });

    it('should match snapshot when selected', () => {
        const { container } = render(<TimelineDot isSelected={true} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot when not selected', () => {
        const { container } = render(<TimelineDot isSelected={false} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
