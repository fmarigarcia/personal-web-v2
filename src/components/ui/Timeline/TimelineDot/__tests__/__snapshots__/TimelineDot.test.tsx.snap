// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineDot should match snapshot when not selected 1`] = `
<div
  class="w-4 h-4 rounded-full border-3 transition-all duration-300 mb-4 bg-white border-blue-600 group-hover:bg-blue-100 group-hover:scale-110"
/>
`;

exports[`TimelineDot should match snapshot when selected 1`] = `
<div
  class="w-4 h-4 rounded-full border-3 transition-all duration-300 mb-4 bg-blue-600 border-blue-600 scale-125 shadow-lg"
/>
`;
